[BACKEND] B-email-verification

bekir96 opened the pull request:
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Description

I have implemented two endpoints to verify mail account. Collaborated with @abdullahyldz on frontend side.

First endpoint is `email-verify/<uidb64>`. This endpoint firstly come up with register side. When customer register to own page, our gmail account send verify link to customer mail. Verify link contains token in  path, and token expire date is 2 days.  

Second endpoint is `user/verify`. This endpoint come up with profile page. When customer enters own profile page if do not verify your account, customer see verify your account section. Then clicks the this section, our gmail account send verify link to customer mail again.

To test our endpoints with @abdullahyldz , he helps me to run frontend side. To test locally, backend is starting at `http://127.0.0.1:1234/` and frontend is starting at `http://127.0.0.1:5000/`.

# Screenshots

![image](https://user-images.githubusercontent.com/44097383/102927476-20c66700-44a8-11eb-8f80-33c778219431.png)
![WhatsApp Image 2020-12-22 at 22 28 36](https://user-images.githubusercontent.com/44097383/102927702-84509480-44a8-11eb-8133-5c5def68f1d3.jpeg)
![WhatsApp Image 2020-12-22 at 22 29 05](https://user-images.githubusercontent.com/44097383/102927714-887cb200-44a8-11eb-8171-98fe6fd29b3f.jpeg)
![image](https://user-images.githubusercontent.com/44097383/102927719-8a467580-44a8-11eb-8ca9-c58d2699d450.png)
![image](https://user-images.githubusercontent.com/44097383/102927723-8b77a280-44a8-11eb-907f-c58970fdca75.png)
![image](https://user-images.githubusercontent.com/44097383/102927732-8ca8cf80-44a8-11eb-9bde-06548b690f26.png)


# Test results

Test results are satisfied above Screenshot section.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Potential problems/Important remarks

All ENV variables are given by Slack.

bekir96 committed:
Verification implementation is started

abdullahyldz committed:
email verification endpoint

abdullahyldz committed:
message code from BE

bekir96 committed:
Verification should be tested

bekir96 committed:
Merge remote-tracking branch 'origin/F-email-verification' into B-email-verification

bekir96 committed:
first test pass

bekir96 committed:
frontend little change

bekir96 committed:
days changed

MehdiSaffar committed:
setup msw with example handler

MehdiSaffar committed:
add /products/homepage/:count mock

MehdiSaffar committed:
add /product/:productId mock

MehdiSaffar committed:
add /search/products mock

abdullahyldz committed:
verify user button is placed

abdullahyldz committed:
Small changes, Invalid response

abdullahyldz committed:
Merge branch 'master' into F-email-verification

abdullahyldz committed:
is_Verified field to ctx, not verified warning

abdullahyldz committed:
antd result shows in verify endpoint

abdullahyldz committed:
Merge remote-tracking branch 'origin/F-email-verification' into F-email-verification

abdullahyldz committed:
profile page verification notifs

abdullahyldz committed:
Merge branch 'F-mock' into F-email-verification

abdullahyldz committed:
random mock data for email verification

abdullahyldz committed:
update of is_verified with successful verification

bekir96 committed:
Update acc_active_email.html

domain changes

abdullahyldz committed:
Merge branch 'master' into F-email-verification

abdullahyldz committed:
is_verified notification not shown for guests

abdullahyldz committed:
bug solve:  not showing notifs. after init

abdullahyldz committed:
Verify button on the alert, update user field

bekir96 committed:
Merge branch 'master' into B-email-verification

bekir96 committed:
settings.py is changed

abdullahyldz committed:
fix missing var.

bekir96 committed:
Merge branch 'F-email-verification' into B-email-verification

bekir96 committed:
last test has passed and on the frontend side bug fix

abdullahyldz committed:
Email verification tests are complete

abdullahyldz committed:
Merge branch 'master' into B-email-verification

omerfdeniz reviewed (approved):
Good job, can merge

abdullahyldz reviewed (approved):
We have manually tested with @bekir96 and verified the correctness of code.
Great job.

umutoksuz commented on ecommerce/backend/getflix/settings.py:
```diff
@@ -26,9 +26,14 @@
+    'EMAIL_HOST': os.environ.get('EMAIL_HOST'),
+    'EMAIL_HOST_PASSWORD': os.environ.get('EMAIL_HOST_PASSWORD'),
+    'EMAIL_HOST_USER': os.environ.get('EMAIL_HOST_USER'),
+    'DEFAULT_FROM_EMAIL': os.environ.get('DEFAULT_FROM_EMAIL')
```
For those, I think we need to configure our backend deploy script and add encrypted version of email password etc. to repo.

umutoksuz replied:
Merging to master after those variables are configured would be better for deployment purposes

umutoksuz commented on ecommerce/backend/getflix/settings.py:
```diff
@@ -48,6 +53,19 @@ def get_secret(setting, secrets=secrets):
+MAILER_EMAIL_BACKEND = get_secret('MAILER_EMAIL_BACKEND')
+EMAIL_HOST = get_secret('EMAIL_HOST')
+EMAIL_HOST_PASSWORD = get_secret('EMAIL_HOST_PASSWORD')
+EMAIL_HOST_USER = get_secret('EMAIL_HOST_USER')
```
These lines are testing purpose? We should delete them before deploying to master branch

bekir96 replied:
No, ENV variables are:
export EMAIL_BACKEND="django.core.mail.backends.smtp.EmailBackend"
export MAILER_EMAIL_BACKEND=$EMAIL_BACKEND
export EMAIL_HOST="smtp.gmail.com"
export EMAIL_HOST_PASSWORD="getflixecommerce"
export EMAIL_HOST_USER="getflix.e.commerce"
export DEFAULT_FROM_EMAIL=$EMAIL_HOST_USER

group should add these to env variables

umutoksuz reviewed (commented):
Code is well. About some minor points I have small comments

MehdiSaffar committed:
Merge branch 'master' into B-email-verification

MehdiSaffar committed:
ignore .env.production

MehdiSaffar committed:
ignore copy_backend_secrets.sh

MehdiSaffar committed:
proper env handling in backend

- add copy_backend_secrets.sh
- refactors those .sh files to run from anywhere
- update Dockerfile to include .env.production

MehdiSaffar committed:
refactor backend_build_prep.sh

MehdiSaffar committed:
unignored copy_backend_secrets.sh

MehdiSaffar commented:
added the env handling part, will need to be tested tomorrow before merging.

@bekir96 has the .env.production and .env.development files that need to be put in `ecommerce/backend`. 

During dev, for DB_HOST you can pass it either as env variable from command line or by putting it in `.env.development`

Example on linux:

```
DB_HOST=111.222.333.444 python manage.py runserver
```

Please do not make any changes to settings.py from now on

MehdiSaffar committed:
fix typo in backend_build_prep.sh

MehdiSaffar committed:
fix deploy-backend.yml and backend_build_prep.sh

bekir96 committed:
HTTP referer is added

bekir96 committed:
Purchase serializer little change

bcetin committed:
[BACKEND] Search Endpoints (#302)

* Initializing search branch

* Update product serializer

* Search/Products Small Fix

* Implementing search/vendors endpoint

* Implementing search/brands

* Implementing Tests For Search

MehdiSaffar committed:
chmod +x backend_build_prep.sh

karalleyna committed:
[ANDROID] Product Fragment & Product View Model & The design of Product Layout  (#282)

* Create RecommenderAdapter
* Create ProductFragment
* Create ProductViewModel
* Add background styles
* Add string values for the new response of product endpoint
* Add new animations
* Add necessary string values
* Add color and font for the design
* Create the layout of view holder of RecommenderAdapter
* Create the layout of the detailed page of the product
* Add icons
* Added for the test
* Add necessary libraries
* Fix the get request for the product
* Create Image Adapter
* Add rating functions and adapters
* Comment out the unnecessary function for categories view model
* Create three background drawables
* Create the layout for the view holder of ImageAdapter
* Add scrolling functionality and background
* Add image two recyclerviews and circle indicator
* Add necessary string values and fonts
* Add the layout for the view holder of CommentAdapter
* Get request from the /product/id endpoint
* Add necessary file and delete unused imports
* Refactor layouts
* Add the data model for the product
* Update the request for "product/{productId}"
* Update the navigation
* Add text views for category and subcategory
* Remove mock data
* Add necessary bindings
* Change the signature of the get request for  "products/homepage/{numberOfProducts}"
* Rereate CommentAdapter from scratch
* Implement RecommenderAdapter from scratch
* Change the logic of the setProductRating
* Add three endpoint request for ProductViewModel and one for CommentAdapter
* Create the layout for CommentAdapter.ViewHolder from scratch
* Add binding and validation of the response
* Update scrolling
* Redesign card_comment,fragment_product and card_product

asliayk committed:
[ANDROID] Connecting Remaining Endpoints for Cart/Address & Design Improvements (#310)

* address cart layout updated
* product endpoint updated
* address form design improvements + fragment animation added
* improved card view design added
* add credit card functionality added
* credit cart item design + delete bug fixed
* address layout edited
* renamed some files + progress bar redesign
* update address + cart fragments added with navigation
* update address layout + necessary parameters
* update requests binded to the forms
* design refactors

bekir96 committed:
Update verify_email.py

bekir96 committed:
Update account.py

bekir96 committed:
Update verify.py

karalleyna committed:
Change size of ic_credit_card

MehdiSaffar committed:
update secrets

MehdiSaffar committed:
test deployment

MehdiSaffar committed:
ignore handlers

MehdiSaffar committed:
make it work in local, fix settings envs

MehdiSaffar committed:
[FRONTEND] Finalize the checkout page (full backend integration) (#306)

Co-authored-by: Özdeniz Dolu <ozdeniz.dolu@boun.edu.tr>
Co-authored-by: Mehdi Saffar <saffar.mehdi3@gmail.com>

MehdiSaffar committed:
commit

MehdiSaffar committed:
deploy frontend for test too

MehdiSaffar committed:
revert on push

MehdiSaffar committed:
Merge branch 'master' into B-email-verification

MehdiSaffar reviewed (approved):
Integrated necessary changes, works perfectly (tested locally and deployed frontend)
